### <a name="systemthreadingtaskstask-no-longer-throw-objectdisposedexception-after-object-is-disposed"></a><span data-ttu-id="ff301-101">System.Threading.Tasks.Task ya no inicia una excepción ObjectDisposedException después de desechar un objeto</span><span class="sxs-lookup"><span data-stu-id="ff301-101">System.Threading.Tasks.Task no longer throw ObjectDisposedException after object is disposed</span></span>

|   |   |
|---|---|
|<span data-ttu-id="ff301-102">Detalles</span><span class="sxs-lookup"><span data-stu-id="ff301-102">Details</span></span>|<span data-ttu-id="ff301-103">A excepción de <xref:System.Threading.Tasks.Task.System%23IAsyncResult%23AsyncWaitHandle>, los métodos <xref:System.Threading.Tasks.Task?displayProperty=name> ya no inician una excepción <xref:System.ObjectDisposedException?displayProperty=name> después de desechar el objeto. Este cambio admite el uso de tareas en caché.</span><span class="sxs-lookup"><span data-stu-id="ff301-103">Except for <xref:System.Threading.Tasks.Task.System%23IAsyncResult%23AsyncWaitHandle>, <xref:System.Threading.Tasks.Task?displayProperty=name> methods no longer throw an <xref:System.ObjectDisposedException?displayProperty=name> exception after the object is disposed.This change supports the use of cached tasks.</span></span> <span data-ttu-id="ff301-104">Por ejemplo, un método puede devolver una tarea almacenada en caché para representar una operación completada en lugar de asignar una nueva tarea.</span><span class="sxs-lookup"><span data-stu-id="ff301-104">For example, a method can return a cached task to represent an already completed operation instead of allocating a new task.</span></span> <span data-ttu-id="ff301-105">Esto no era posible en versiones anteriores de .NET Framework, ya que cualquier consumidor de la tarea podía desecharla, lo que hacía que se volviera inutilizable.</span><span class="sxs-lookup"><span data-stu-id="ff301-105">This was impossible in previous .NET Framework versions, because any consumer of the task could dispose of it, which rendered it unusable.</span></span>|
|<span data-ttu-id="ff301-106">Sugerencia</span><span class="sxs-lookup"><span data-stu-id="ff301-106">Suggestion</span></span>|<span data-ttu-id="ff301-107">Tenga en cuenta que es posible que los métodos Task ya no inicien excepciones <xref:System.ObjectDisposedException?displayProperty=name> cuando el objeto se desecha.</span><span class="sxs-lookup"><span data-stu-id="ff301-107">Be aware that Task methods may no longer throw <xref:System.ObjectDisposedException?displayProperty=name> in cases when the object is disposed.</span></span> <span data-ttu-id="ff301-108">Si una aplicación dependía de esta excepción para conocer que se desechó una tarea, se debería actualizar para comprobar de forma explícita el estado de la tarea mediante <xref:System.Threading.Tasks.Task.Status>.</span><span class="sxs-lookup"><span data-stu-id="ff301-108">If an app was depending on this exception to know that a task was disposed, it should be updated to explicitly check the task's status using <xref:System.Threading.Tasks.Task.Status>.</span></span>|
|<span data-ttu-id="ff301-109">Ámbito</span><span class="sxs-lookup"><span data-stu-id="ff301-109">Scope</span></span>|<span data-ttu-id="ff301-110">Secundaria</span><span class="sxs-lookup"><span data-stu-id="ff301-110">Minor</span></span>|
|<span data-ttu-id="ff301-111">Versión</span><span class="sxs-lookup"><span data-stu-id="ff301-111">Version</span></span>|<span data-ttu-id="ff301-112">4.5</span><span class="sxs-lookup"><span data-stu-id="ff301-112">4.5</span></span>|
|<span data-ttu-id="ff301-113">Tipo</span><span class="sxs-lookup"><span data-stu-id="ff301-113">Type</span></span>|<span data-ttu-id="ff301-114">Tiempo de ejecución</span><span class="sxs-lookup"><span data-stu-id="ff301-114">Runtime</span></span>|

